runtime/src/gpu/common/rocm-version.h

```diff
@@ -0,0 +1,35 @@
+
+/*
+ * Copyright 2020-2023 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.  *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __HIP_PLATFORM_AMD__
+#define __HIP_PLATFORM_AMD__
+#endif
+#include <hip/hip_common.h>
+#include <hip/hip_runtime.h>
+
+#if __has_include(<rocm-core/rocm_version.h>)  // 5.x wants this
+#include <rocm-core/rocm_version.h>
+#elif __has_include(<rocm/rocm_version.h>)  // 4.x wants this
+#include <rocm/rocm_version.h>
+#elif __has_include(<rocm_version.h>)  // Deprecated. 5.x used to want this
+#include <rocm_version.h>
+#elif !defined(ROCM_VERSION_MAJOR)
+#define ROCM_VERSION_MAJOR 4 // this is the safe bet
+#endif
```

Comment:
If I understand this right, we are searching headers for `ROCM_VERSION_MAJOR`, and if its not set defaulting to 4? So should `#elif !defined(ROCM_VERSION_MAJOR)` still be apart of the elif chain?

Reply:
That's right. But I am not following... which part of what I have feels wrong? Would you expect this to be just `else`? If that's the case, what I have feels safer in case:

1. this file was included multiple times, or,
2. `ROCM_VERSION_MAJOR` was set via other means like a `-D` or something.

(2) is not a big worry obviously, but (1) is.

Reply:
I would have expected something like
```c++
#if __has_include(<rocm-core/rocm_version.h>)  // 5.x wants this
#include <rocm-core/rocm_version.h>
#elif __has_include(<rocm/rocm_version.h>)  // 4.x wants this
#include <rocm/rocm_version.h>
#elif __has_include(<rocm_version.h>)  // Deprecated. 5.x used to want this
#include <rocm_version.h>
#endif

#if !defined(ROCM_VERSION_MAJOR)
#define ROCM_VERSION_MAJOR 4 // this is the safe bet
#endif
```

Reply:
Oh, I see. OK, yeah I like that better.

Reply:
https://github.com/chapel-lang/chapel/pull/24018/commits/79fbc96d07e83e0e7a0f28e795cdf5d4b1a69880